server/src/test/java/org/candlepin/pinsetter/tasks/PopulateHostedDBTaskTest.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.pinsetter.tasks;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+
+import org.candlepin.model.Content;
+import org.candlepin.model.ContentCurator;
+import org.candlepin.model.PoolCurator;
+import org.candlepin.model.Product;
+import org.candlepin.model.ProductContent;
+import org.candlepin.model.ProductCurator;
+import org.candlepin.service.ProductServiceAdapter;
+import org.candlepin.test.DatabaseTestFixture;
+import org.candlepin.test.TestUtil;
+
+import org.junit.Test;
+import org.quartz.JobExecutionContext;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Map.Entry;
+
+
+
+/**
+ * PopulateHostedDBTaskTest
+ */
+public class PopulateHostedDBTaskTest extends DatabaseTestFixture {
+
+    @Test
+    @SuppressWarnings("checkstyle:methodlength")
+    public void testExecute() throws Exception {
+        // Setup
+        JobExecutionContext jec = mock(JobExecutionContext.class);
+
+        PoolCurator poolCuratorSpy = spy(this.poolCurator);
+        ProductServiceAdapter psa = mock(ProductServiceAdapter.class);
+
+        HashSet<String> productIds = new HashSet<String>();
+        productIds.add("prod1");
+        productIds.add("prod2");
+        productIds.add("prod3");
+
+        HashSet<String> dependentProductIds1 = new HashSet<String>();
+        dependentProductIds1.add("dprod1");
+        dependentProductIds1.add("dprod2");
+        dependentProductIds1.add("prod3");
```

Comment:
That whole thing is to make sure it doesn't keep looping around asking for the same products repeatedly.
Related to your question above, the five collections are to support the three collections I want, as well as two extra collections that will be used only if it's not filtering products it's already seen.